pkg/systemd/generate/pods.go

```diff
@@ -242,7 +242,13 @@ func generatePodInfo(pod *libpod.Pod, options entities.GenerateSystemdOptions) (
 		nameOrID = pod.Name()
 		ctrNameOrID = infraCtr.Name()
 	}
-	serviceName := fmt.Sprintf("%s%s%s", options.PodPrefix, options.Separator, nameOrID)
+
+	var serviceName string
+	if options.PodPrefix == "" {
+		serviceName = nameOrID
+	} else {
```

Comment:
same comment about reducing this down to one conditional

Reply:
Updated the code to use a function which has a single if condition